src/ML/src/MANNAutoregressive.cpp

```diff
@@ -663,54 +721,32 @@ bool MANNAutoregressive::advance()
         return false;
     }
 
-    // update the support foot ptr for the next iteration
-    if (supportFootPtr != m_pimpl->supportFootPtr && !supportFootPtr->estimatedContactPtr->isActive)
-    {
-        supportFootPtr->estimatedContactPtr->isActive = true;
-        supportFootPtr->estimatedContactPtr->switchTime = m_pimpl->currentTime;
-        supportFootPtr->estimatedContactPtr->pose = BipedalLocomotion::Conversions::toManifPose(
-            m_pimpl->kinDyn.getWorldTransform(supportFootPtr->discreteGeometryContact.index));
-    }
-
-    // Check if the contact has been deactivated.
-    // In this context we have to check if difference of the height of the lower corner of the two
-    // feet is greater then a given threshold.
+    // store the output
     const double leftFootHeight
         = m_pimpl->kinDyn.getWorldTransform(m_pimpl->leftFoot.discreteGeometryContact.index)
               .getPosition()[2];
     const double rightFootHeight
         = m_pimpl->kinDyn.getWorldTransform(m_pimpl->rightFoot.discreteGeometryContact.index)
               .getPosition()[2];
-    constexpr double tolerance = 0.005;
-    if (std::abs(leftFootHeight - rightFootHeight) > tolerance)
-    {
-        // if the support foot is not the left and the left is active, then we deactivate the left
-        if (supportFootPtr->estimatedContactPtr != &m_pimpl->output.leftFoot
-            && m_pimpl->output.leftFoot.isActive)
-        {
-            m_pimpl->output.leftFoot.isActive = false;
-        }
-        // if the support foot is not the right and the right is active, then we deactivate the
-        // right
-        else if (supportFootPtr->estimatedContactPtr != &m_pimpl->output.rightFoot
-                 && m_pimpl->output.rightFoot.isActive)
-        {
-            m_pimpl->output.rightFoot.isActive = false;
-        }
-    }
 
-    // store the output
-    m_pimpl->output.leftFoot.lastUpdateTime = m_pimpl->currentTime;
-    m_pimpl->output.rightFoot.lastUpdateTime = m_pimpl->currentTime;
+    m_pimpl->updateContact(-leftFootHeight,
+                           m_pimpl->leftFootSchmittTrigger,
+                           m_pimpl->output.leftFootSchmittTriggerState,
+                           m_pimpl->output.leftFoot);
+    m_pimpl->updateContact(-rightFootHeight,
```

Comment:
I would sugggest adding a contact to explain why the minus

Reply:
Done here: https://github.com/ami-iit/bipedal-locomotion-framework/commit/4634f98306381828c9eb9bf471e4402a3f970ad2